### PR TITLE
GoogleNet modern-cnn implementation.

### PR DESCRIPTION
**Pull request by aksrajvanshi**
Adding GoogleNet implementation ipynb. (Section 7.4)

### CODE REVIEW

File: jupyter/d2l-java/chapter_convolutional-modern/googlenet.ipynb [hunk 1]
```diff
@@ -0,0 +1,668 @@
...
+    "            SequentialBlock p4 = new SequentialBlock()\n",
+    "                    .add(Pool.maxPool2dBlock(new Shape(3, 3), new Shape(1, 1), new Shape(1, 1)))\n",
+    "                    .add(Conv2d.builder()\n",
+    "                            .setFilters(c4)\n",
+    "                            .setKernelShape(new Shape(1, 1))\n",
+    "                            .build())\n",
+    "                    .add(Activation::relu);\n",
+    "\n",
+    "            // Concatenate the outputs on the channel dimension\n",
+    "            return new SequentialBlock()\n",
```
Comment (lanking520) [inline jupyter/d2l-java/chapter_convolutional-modern/googlenet.ipynb:206]: InceptionBlock itself is a parallelblock, why using SequentialBlock to wrap it?

Comment (aksrajvanshi) [reply jupyter/d2l-java/chapter_convolutional-modern/googlenet.ipynb]: Makes sense. Made the function return a Block type in the declaration and just returning a Parallel Block object.

File: jupyter/d2l-java/chapter_convolutional-modern/googlenet.ipynb [hunk 2]
```diff
@@ -0,0 +1,662 @@
...
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "1"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "public class Inception {\n",
```
Comment (lanking520) [inline jupyter/d2l-java/chapter_convolutional-modern/googlenet.ipynb:146]: It doesn't make sense to create a class with only one function and no additional constructor. Is that possible to just expose InceptionBlock instead? And use return type `ParallelBlock`

Comment (aksrajvanshi) [reply jupyter/d2l-java/chapter_convolutional-modern/googlenet.ipynb:146]: Sure! I was a little skeptical about it too, but was trying to follow the book in terms of class structure.

File: jupyter/d2l-java/chapter_convolutional-modern/googlenet.ipynb [hunk 3]
```diff
@@ -0,0 +1,659 @@
...
+    "      :cite:`Szegedy.Vanhoucke.Ioffe.ea.2016`.\n",
+    "    * Include it in the residual connection\n",
+    "      :cite:`Szegedy.Ioffe.Vanhoucke.ea.2017`, as described later in\n",
+    "      :numref:`sec_resnet`.\n",
+    "1. What is the minimum image size for GoogLeNet to work?\n",
+    "1. Compare the model parameter sizes of AlexNet, VGG, and NiN with GoogLeNet. How do the latter two network architectures significantly reduce the model parameter size?\n",
+    "1. Why do we need a large range convolution initially?\n",
+    "\n",
+    "\n",
+    "## [Discussions](https://discuss.mxnet.io/t/2357)\n",
```
Comment (lanking520) [inline jupyter/d2l-java/chapter_convolutional-modern/googlenet.ipynb:636]: Remove this?

**Pull request merged**